components/badge/index.css

```
@@ -68,6 +68,8 @@
 	--spectrum-badge-icon-spacing-horizontal: var(--spectrum-component-edge-to-visual-100);
 	--spectrum-badge-icon-spacing-vertical-top: var(--spectrum-component-top-to-workflow-icon-100);
 	--spectrum-badge-icon-only-spacing-horizontal: var(--spectrum-component-edge-to-visual-only-100);
+
+	--highcontrast-spectrum-badge-border-color: CanvasText;
```

Update needed here as well :)

```suggestion
	--highcontrast-badge-border-color: CanvasText;
```